pkg/api/globalquery/client/client.go

```diff
@@ -59,3 +90,50 @@ func (c *Client) Query(ctx context.Context, args *query.Args) (*results.Result,
 
 	return res, nil
 }
+
+// Query performs the global query and returns its result, while consuming updates to partial results
+func (sse *SSEClient) Query(ctx context.Context, args *query.Args) (*results.Result, error) {
+	logger := logging.FromContext(ctx)
+
+	if sse.onFinish == nil || sse.onUpdate == nil {
+		return nil, errors.New("no event callbacks provided (onUpdate, onFinish)")
+	}
+
+	// use a copy of the arguments, since some fields are modified by the client
+	queryArgs := *args
+
+	// whatever happens, the results are expected to be returned in json
+	queryArgs.Format = "json"
```

Comment:
`const` for the format(s) maybe (I think we're using those in several places already)?